tests/smoke.rs

```diff
@@ -143,6 +152,12 @@ fn check_copy_and_move() {
         with_maildir("maildir2", |maildir2| {
             let id = "1463868505.38518452d49213cb409aa1db32f53184";
 
+            // check that we cannot copy a message from and to the same maildir
+            assert_eq!(
+                maildir1.copy_to(id, &maildir1).unwrap_err().kind(),
+                std::io::ErrorKind::InvalidInput,
+            );
+
             // check that the message is present in maildir1 but not in maildir2
             assert!(maildir1.find(id).is_some());
             assert!(maildir2.find(id).is_none());
```

Comment:
Actually it would also be good to ensure the file is not empty, since the failure mode of the self-copy would be to leave the file in place but make it zero length. You could call `.date()` on the `MailEntry` to ensure it has some valid data.

Reply:
I'm not so convince about this one. Testing that the file is not empty means testing the `std::fs::rename` function, which is useless in my opinion. But if you really want, I can implement it!

Reply:
I understand better. Thanks to have done the changes!